globus_cli/safeio/errors.py

```diff
@@ -35,17 +35,18 @@ def __str__(self):
                 name, spacer, spacer.join(self.value.split('\n')))
 
 
-def write_error_info(error_name, fields):
-    message = None
+def write_error_info(error_name, fields, message=None):
     if outformat_is_json():
         # dictify joined tuple lists and dump to json string
         message = json.dumps(
             dict(
                 [('error_name', error_name)] +
                 [(f.name, f.value) for f in fields]),
             indent=2)
-    else:
-        message = 'A {0} Occurred.\n{1}'.format(
+
+    if not message:
+        message = 'A{0} {1} Occurred.\n{2}'.format(
+            "n" if error_name[0] in "aeiouAEIOU" else "",
```

Comment:
Hahaha! I would probably have just tried to avoid the indefinite article, but this works too.